### PR TITLE
Dump Pod as YAML in logs for KubernetesPodOperator

### PR DESCRIPTION
**Pull request by mik-laj**
CC: @kuromt 

The YAML format is the more common representation for k8s objects.

---
Make sure to mark the boxes below before creating PR: [x]

- [X] Description above provides context of the change
- [X] Unit tests coverage for changes (not needed for documentation changes)
- [X] Target Github ISSUE in description if exists
- [X] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [X] Relevant documentation is updated including usage instructions.
- [X] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py [hunk 1]
```diff
@@ -378,7 +379,7 @@ def create_new_pod_for_operator(self, labels, launcher) -> Tuple[State, k8s.V1Po
         )
 
         self.pod = pod
-        self.log.info("Starting pod %s", pod)
+        self.log.info("Starting pod:\n%s", yaml.safe_dump(pod.to_dict()))
```
Comment (kaxil) [inline airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py]: Are there any security implications of doing this? Exposing secrets etc in the logs

Comment (mik-laj) [reply airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py]: By design, only metadata and a declarative configuration description should be included here. Secrets should be in K8s Secrets. 
See: https://kubernetes.io/docs/concepts/configuration/secret/

Comment (kaxil) [reply airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py]: I would still increase the log level as DEBUG i.e. you would only want the entire pod spec when you are debugging something unless there is a use-case which I have not thought off

Comment (mik-laj) [inline airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py]:
```suggestion
        self.log.debug("Starting pod:\n%s", yaml.safe_dump(pod.to_dict()))
```

Comment (mik-laj) [reply airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py]: Good idea. I updated PR.

**Review by kaxil — APPROVED** [review]

File: kubernetes_tests/test_kubernetes_pod_operator.py [hunk 2]
```diff
@@ -739,7 +740,20 @@ def test_pod_template_file(
         )
         monitor_mock.return_value = (State.SUCCESS, None)
         context = create_context(k)
-        k.execute(context)
+        with self.assertLogs(k.log) as cm:
+            k.execute(context)
+            expected_line = textwrap.dedent("""\
+            INFO:airflow.task.operators:Starting pod:
```
Comment (mik-laj) [inline kubernetes_tests/test_kubernetes_pod_operator.py]:
```suggestion
            DEBUG:airflow.task.operators:Starting pod:
```

**Pull request merged**